src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -535,18 +541,172 @@ export const maskUidOnLastField = (
+      getFormattedFunction(row, hashedFields),
+    )
+  } else if (isProcessedCheckboxResponse(response)) {
+    const checkbox = getAnswerForCheckbox(response)
+    return [getFormattedFunction(checkbox, hashedFields)]
+  } else {
+    return [getFormattedFunction(response, hashedFields)]
+  }
+}
+
+/**
+ * Helper function to mask NRICs in Corppass Validated UID
+ */
+const maskField = (field: EmailAutoReplyField): EmailAutoReplyField => {
+  const maskedAnswerTemplate = field.answerTemplate.map((answer) => {
+    return answer.length >= 4 // defensive, in case UID length is less than 4
+      ? '*'.repeat(answer.length - 4) + answer.substr(-4)
+      : answer
+  })
+  return {
+    question: field.question,
+    answerTemplate: maskedAnswerTemplate,
+  }
+}
+
+/**
+ * Function to extract information for email json field from response
+ * Json field is used for data collation tool
+ */
+const getJsonFormattedResponse = (
+  response: ResponseFormattedForEmail,
+): EmailJsonField | undefined => {
+  const { answer, fieldType } = response
+  // Headers are excluded from JSON data
+  if (fieldType !== BasicField.Section) {
+    const jsonData = {
+      question: getJsonPrefixedQuestion(response),
+      answer,
+    }
+    return jsonData
+  }
+  return undefined
+}
+
+/**
+ * Function to extract information for email form field from response
+ * Form field is used to send responses to admin
+ */
+const getFormFormattedResponse = (
+  response: ResponseFormattedForEmail,
+  hashedFields: Set<string>,
+): EmailFormField => {
+  const { answer, fieldType } = response
+  const answerSplitByNewLine = answer.split('\n')
+  const formData = {
+    question: getFormDataPrefixedQuestion(response, hashedFields),
+    answerTemplate: answerSplitByNewLine,
+    answer,
+    fieldType,
+  }
+  return formData
+}
+
+/**
+ * Function to extract information for email autoreply field from response
+ * Autoreply field is used to send confirmation emails
+ */
+const getAutoReplyFormattedResponse = (
+  response: ResponseFormattedForEmail,
+): EmailAutoReplyField | undefined => {
+  const { question, answer, isVisible } = response
+  const answerSplitByNewLine = answer.split('\n')
+  // Auto reply email will contain only visible fields
+  if (isVisible) {
+    const autoReplyData = {
+      question, // No prefixes for autoreply
+      answerTemplate: answerSplitByNewLine,
+    }
+    return autoReplyData
+  }
+  return undefined
+}
+
+export class EmailDataObj {
+  parsedResponses: ProcessedFieldResponse[]
+  hashedFields: Set<string>
+  authType: AuthType
+
+  constructor(
+    parsedResponses: ProcessedFieldResponse[],
+    hashedFields: Set<string>,
+    authType: AuthType,
+  ) {
+    this.parsedResponses = parsedResponses
+    this.hashedFields = hashedFields
+    this.authType = authType
+  }
+
+  /**
+   * Getter function to return jsonData which is used for data collation tool
+   */
+  get jsonData(): EmailJsonField[] {
```

Comment:
could we pls change the naming convention of these variables? `jsonData` is very abstruse - maybe `collationToolData`?

Reply:
changed to `dataCollationData`, but for `formData` and `autoReplyData`, I've left them because I think this probably requires a separate issue on its own? both terms are used throughout the codebase in other areas - esp formData which is also used in encrypt submission route

Reply:
you could just do a global rename of the variable (F2 in VSCode). this is low risk (if something goes wrong it won't compile) and easy to review as well.

Reply:
as discussed, can be done separately

Reply:
as discussed on slack, to kiv as global rename doesn't work cleanly in this case